### PR TITLE
Use Faster FSM

### PR DESCRIPTION
**Pull request by lapp0**

### CODE REVIEW

File: tests/generate/test_integration_vllm.py [hunk 1]
```diff
@@ -1,239 +0,0 @@
...
-           "bar": {"type": "string", "maxLength": 4}
-        },
-      "required": ["foo", "bar"]
-      }
-    """
-
-    sampling_params = SamplingParams(temperature=0)
-    result = generate.json(model, schema, whitespace_pattern="")(
-        prompt, max_tokens=100, seed=10, sampling_params=sampling_params
-    )
```
Comment (rlouf) [inline tests/generate/test_integration_vllm.py:227]: Why are we removing the `vLLM` integration tests in this PR?

File: outlines/fsm/guide.py [hunk 2]
```diff
@@ -127,64 +74,15 @@ def copy(self):
...
+    return uncached_create_states_mapping(regex_string, tokenizer)
 
-    return states_to_token_maps, empty_token_ids, regex_fsm.finals
 
-
-class RegexGuide(Guide):
-    """Guide to generate text in the language of a regular expression."""
-
-    initial_state = 0
+class RegexGuide(CoreRegexGuide):
```
[the remark this answers is not in the document]
Comment (lapp0) [reply outlines/fsm/guide.py:81]: This is the cached implementation of `RegexGuide`

File: tests/generate/test_integration_vllm.py (hunk 1 above)
Comment (lapp0) [reply tests/generate/test_integration_vllm.py:227]: This was a mistake, reverted.

**Comment by rlouf** [comment]
LGTM!

**Pull request merged**